### PR TITLE
[REF] stock*,purchase*,mrp: remove some related store

### PR DESCRIPTION
**Pull request by ryv-odoo**
[REF] stock*,purchase*,mrp: remove store of some related fields

Since https://github.com/odoo/odoo/pull/127353, we can groupby/aggregate/sort on non-store related field.
Remove store=True of some related fields (tradeoff is taken in account) for every logistics modules.

[REM] mail: remove unused field alias_domain_name

https://github.com/odoo/enterprise/pull/58293
https://github.com/odoo/upgrade/pull/5793

### CODE REVIEW

**Review by amoyaux — APPROVED** [review]

File: addons/stock/models/stock_lot.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def _read_group_location_id(self, locations, domain):
         required=True, check_company=True)
     product_uom_id = fields.Many2one(
         'uom.uom', 'Unit of Measure',
-        related='product_id.uom_id', store=True)
+        related='product_id.uom_id')
```
Comment (KangOl) [inline addons/stock/models/stock_lot.py:51]: Shouldn't this one be kept, for tracking purpose?

Comment (ryv-odoo) [reply addons/stock/models/stock_lot.py:51]: What do you mean ? If the source change, `product_uom_id` will be updated too. And both fields doesn't used tracking anyway (and even if they do, we don't want to repeat the information on every lots).

**Comment by ryv-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by ryv-odoo** [comment]
@robodoo r+

**Pull request closed without merging**